src/org/opendatakit/briefcase/export/ExportForms.java

```diff
@@ -140,7 +142,12 @@ public ExportConfiguration getConfiguration(String formId) {
   }
 
   public void removeConfiguration(FormStatus form) {
-    customConfigurations.remove(getFormId(form));
```

Comment:
Wow, this change makes this method a liar! We can't do this... You can change the implementation, but not the meaning or the intended behavior. Here we want to **remove** a configuration, not empty its contents.

Reply:
@ggalmazor with this changes a form config is never empty and contains at least one field which is the formNeedsPrivateKey which is always either true or false and this cant be changed by the user. So technicaly this field can not be edited by the user while the others can be. This brings in the need to verify if the fields that can be edited by the user are empty (isUserConfigsEmpty()). Using this logic when a user clears a config removing the config totally will go against the logic. So I went ahead and instead cleared all the fields except the formNeedsPrivateKey field which will be required to display the export button as disabled/enabled. But the previous functionality of the clearall button before this change is not changed just the logic. I agree the removeConfig method should do what it refers to, but in this case I might change it to clearConfig since we never remove a config technically. We might also decide to remove configs for a form when the user clicks on the press all button. But this means the form config will depend on the default configuration upon export

Reply:
In order to make the smallest possible change, I'd propose this:
- if the user removes the configuration, we just remove it. That's one thing.
- The other thing is how we decide that a conf is empty. To do that, we add a new condition following this criteria:
  - if it needs pem file and the pemfile is empty, then the conf is empty
  - if it needs pem file and the pemfile is not empty, then the conf is not empty
  - If it doesn't need pem file, then the conf is not empty